ui-v2/app/mixins/with-resizing.js

```diff
@@ -0,0 +1,27 @@
+import Mixin from '@ember/object/mixin';
+import { get } from '@ember/object';
+export default Mixin.create({
+  win: window,
+  init: function() {
+    this._super(...arguments);
+    this.handler = e => {
+      const win = e.target;
+      this.resize({
+        detail: { width: win.innerWidth, height: win.innerHeight },
+      });
+    };
+  },
+  resize: function(e) {},
+  didInsertElement: function() {
+    this._super(...arguments);
+    get(this, 'win').addEventListener('resize', this.handler);
+    this.didAppear();
+  },
+  didAppear: function() {
+    this.handler({ target: get(this, 'win') });
+  },
+  willDestroyElement: function() {
+    get(this, 'win').removeEventListener('resize', this.handler);
+    this._super(...arguments);
+  },
+});
```

Comment:
Nice mixin. One suggestion I have is to put emphasis on resize in some manner to make it obvious that that is the interesting part for a consumer of this mixin. This could be as simple as moving `resize` to the first thing in the file so it's hard to overlook, or something more aggressive [like this](https://github.com/hashicorp/nomad/blob/280d6a3f5536b98f52b7c44cffda82a41305b77d/ui/app/mixins/window-resizable.js#L7-L9), which throws an error if a component/controller/whatever is mixing in the mixin without overriding the expected method.

Reply:
Done